modules/transactions/src/main/java/org/apache/ignite/internal/tx/storage/state/rocksdb/TxStateRocksDbStorage.java

```diff
@@ -243,13 +243,19 @@ public TxStateRocksDbStorage(
         }
 
         try {
-            RocksIterator rocksIterator = db.newIterator();
+            byte[] lowerBound = ByteBuffer.allocate(Short.BYTES + 1).putShort((short) partitionId).put(new byte[1]).array();
+            byte[] upperBound = ByteBuffer.allocate(Short.BYTES).order(ByteOrder.BIG_ENDIAN).putShort((short) (partitionId + 1)).array();
```

Comment:
These two could be fields

Reply:
scan seems not to be frequent operation, meanwhile there could be huge amount of tx state storage instances